Automattic/wp-calypso
# Build: Move babel.config.js to calypso-build package

#### Changes proposed in this Pull Request

The idea I'm trying to implement here is to move the 'generic' part of `babel.config.js` to `packages/calypso-build/`, and keeping the Calypso-specific ones in a root level `babel.config.js`. I'm also moving the Jetpack blocks specific stuff to a `.babelrc.js` in `packages/jetpack-blocks/`.

Note that there are a few caveats with a `.babelrc`-based config. [Notably](https://babeljs.io/docs/en/config-files#file-relative-configuration):

> Searching will stop once a directory containing a `package.json` is found, so a relative config only applies within a single package.

Futher reading: I found that [this doc](https://babeljs.io/docs/en/config-files) is pretty relevant. Took me a while to grok it though :sweat_smile: 

#### Testing instructions

```
npm run distclean
npm ci
npm start
```

and verify in your browser that Calypso actually works.

~Then, build Jetpack blocks~

Blocks are gone. Removed.

Finally, test package:

```
npm run test-packages
```

TODO:
- [ ] Make `jetpack-blocks` package (and others) pick up the right babel config
  - Need to set `TranspileConfig`'s `configFile` option from `package/jetpack-blocks/bin/build.js`, _or_ to make our webpack/babel setup automatically respect `.babelrc.js` (Tho I think explicit would be better than implicit.)
